### PR TITLE
Allow setting attribute weights per attribute set.

### PR DESCRIPTION
**Pull request by nikunjkotecha**

### CODE REVIEW

**Comment by nikunjkotecha** [comment]
cc @sdelbosc @vbouchet31

**Comment by malachy-mcconnell** [comment]
Hi Nikunj, there are some nice ode changes here but we expect attribute weights to by synchronized from the ecommerce system and not set in Drupal. Do you agree? Is this a quick-fix until we can implement synchronisation of attribute weights (sort order)?

**Comment by nikunjkotecha** [comment]
Hi @malachy-mcconnell, I checked the code around automated synchronisation of attribute weights. But that still doesn't support multiple attribute (like the old code). I updated the code around extractConfigurableOptions too in the PR and updated logic should work as is. Probably we may want to set default value in config to `attribute_weights: {  }` as not all systems would have size, otherwise everything here works.

Sortable weights is something we will need to define - reason behind that - we may not have all the attributes in taxonomy terms and it is a performance killer which we should avoid till really required in the project.

File: modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php [hunk 1]
```diff
@@ -203,7 +203,10 @@ public static function configurableAjaxCallback(array $form, FormStateInterface
     $dynamic_parts = &$form['ajax'];
 
     $configurables = $form_state->getValue('configurables');
-    $configurable_form_settings = \Drupal::service('config.factory')->get('acm_sku.configurable_form_settings');
+
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Comment (miromichalicka) [inline modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php]: Inject through DI

File: modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php [hunk 2]
```diff
@@ -219,13 +222,9 @@ public static function configurableAjaxCallback(array $form, FormStateInterface
     }
 
     if ($tree_pointer instanceof SKU) {
-      $plugin = $tree_pointer->getPluginInstance();
-
-      $view_builder = \Drupal::entityTypeManager()
-        ->getViewBuilder('acm_sku');
+      $view_builder = \Drupal::entityTypeManager()->getViewBuilder('acm_sku');
```
Comment (miromichalicka) [inline modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php]: Inject through DI

File: modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php [hunk 3]
```diff
@@ -493,7 +490,12 @@ public function deriveProductTree(SKU $sku) {
       $tree['configurables'][$configurable['code']] = $configurable;
     }
 
-    $configurable_weights = \Drupal::service('config.factory')->get('acm_sku.configurable_form_settings')->get('attribute_weights');
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Comment (miromichalicka) [inline modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php]: Inject through DI

File: modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php [hunk 4]
```diff
@@ -664,23 +666,25 @@ public function cartName(SKU $sku, array $cart, $asString = FALSE) {
...
    *
+   * @param string $attribute_set
+   *   Attribute set.
    * @param array $configurable_options
    *   Array with configurable options.
    */
-  protected function extractConfigurableOptions(array $configurable_options) {
+  protected function extractConfigurableOptions($attribute_set, array $configurable_options) {
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Comment (miromichalicka) [inline modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php]: Inject through DI

File: modules/sku/src/Plugin/AcquiaCommerce/SKUType/Simple.php [hunk 5]
```diff
@@ -27,7 +27,8 @@ public function addToCartForm(array $form, FormStateInterface $form_state, SKU $
       return $form;
     }
 
-    $configurable_form_settings = \Drupal::service('config.factory')->get('acm_sku.configurable_form_settings');
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Comment (miromichalicka) [inline modules/sku/src/Plugin/AcquiaCommerce/SKUType/Simple.php]: Inject through DI

File: modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php [hunk 6]
```diff
@@ -45,8 +45,13 @@ public function addToCartForm(array $form, FormStateInterface $form_state, SKU $
     ];
 
     $configurables = unserialize($sku->field_configurable_attributes->getString());
-    $configurable_form_settings = \Drupal::service('config.factory')->get('acm_sku.configurable_form_settings');
-    $configurable_weights = $configurable_form_settings->get('attribute_weights');
+
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Comment (miromichalicka) [inline modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php]: Inject through DI

**Review by miromichalicka — CHANGES_REQUESTED** [review]

File: modules/sku/src/Plugin/AcquiaCommerce/SKUType/Simple.php (hunk 5 above)
Comment (nikunjkotecha) [reply modules/sku/src/Plugin/AcquiaCommerce/SKUType/Simple.php]: @miromichalicka I request to be done latest as part of separate ticket, I don't have a way to test this as of now as we are still on 1.0 which doesn't use dependency injection at all

cc @sdelbosc

**Comment by malachy-mcconnell** [comment]
On hold. Do not merge until the requested Dependency Injection can be added.

**Comment by nikunjkotecha** [comment]
fixed conflicts

**Pull request merged**